src/UnityExtension/Assets/Editor/GitHub.Unity/ExtensionLoader/UnityAPIWrapper.cs

```diff
@@ -0,0 +1,20 @@
+using UnityEditor;
+using UnityEngine;
+using System.IO;
+using System;
+
+namespace GitHub.Unity
+{
+    [InitializeOnLoad]
+    public class UnityAPIWrapper : ScriptableSingleton<UnityAPIWrapper>
+    {
+        static UnityAPIWrapper()
+        {
+#if UNITY_2018_2_OR_NEWER
```

Comment:
Where or how is this preprocessor directive set?

Reply:
That's Unity setting it, every Unity version sets defines so you can conditionally compile stuff

Reply:
I'm still confused, I thought preprocessor directives get "compiled out". So if we compile and distribute with unity 5.6, and the directive stays (?) then Unity 2018.2 recompiles on first load with the flag set?

Reply:
Okay, so to answer my own question. Yes, of course preprocessor directives get compiled out.

The contents of `ExtensionLoader` are distributed directly, not compiled. So the `.cs` files get compiled in the target Unity instance where the preprocessor directives get set as expected.